File: src/agents/BaseAgent.ts
import { IAgent, AgentConfig, AgentResponse } from './types';

export abstract class BaseAgent implements IAgent {
    id: string;
    config: AgentConfig;

    constructor(id: string, config: AgentConfig) {
        this.id = id;
        this.config = config;
    }

    abstract execute(task: string, context?: any): Promise<AgentResponse>;

    protected log(message: string) {
        console.log(`[Agent ${this.config.name}]: ${message}`);
    }
}

File: backend/agents/base_agent.py
from abc import ABC, abstractmethod
from typing import Any, Dict, Optional

class BaseAgent(ABC):
    def __init__(self, name: str, description: str):
        self.name = name
        self.description = description

    @abstractmethod
    def execute(self, task: str, context: Optional[Dict[str, Any]] = None) -> Dict[str, Any]:
        """
        Execute the agent's task.
        """
        pass

    def log(self, message: str):
        print(f"[{self.name}]: {message}")
